modules/pricer/src/main/java/com/opengamma/strata/pricer/rate/future/AbstractIborFutureTradePricer.java

```diff
@@ -48,5 +51,27 @@ public CurrencyAmount presentValue(IborFutureTrade trade, double currentPrice, d
     double pv = (priceIndex - referenceIndex) * trade.getQuantity();
     return CurrencyAmount.of(future.getCurrency(), pv);
   }
+  
+  /**
+   * Calculates the reference price for a futures trade. The reference price is the trade price before any margining 
+   * has taken place and the price used for the last margining otherwise.
+   * 
+   * @param trade  the trade to price
+   * @param valuationDate  the date for which the reference price should be calculated
+   * @param lastMarginPrice  the last price used in the margining 
+   * @return the reference price
+   */
+  public double referencePrice(IborFutureTrade trade, LocalDate valuationDate, double lastMarginPrice) {
+    ArgChecker.notNull(valuationDate, "valuation date");
+    Optional<LocalDate> tradeDate = trade.getTradeInfo().getTradeDate();
+    ArgChecker.isTrue(tradeDate.isPresent(), "trade date should be populated");
+    double referencePrice;
```

Comment:
In cases like this, it is better to avoid the local variable and just return from each part of the if statement. Or use a ternary operator `(a ? b : c)`